ics_space_amplification_test.py

```diff
@@ -0,0 +1,218 @@
+import math
+import time
+
+from longevity_ics_test import IcsLongevetyTest
+from sdcm.cluster import SCYLLA_DIR
+from sdcm.utils.common import retrying
+
+KB_SIZE = 2 ** 10
+MB_SIZE = KB_SIZE * 1024
+GB_SIZE = MB_SIZE * 1024
+MAX_ICS_SPACE_AMPLIFICATION_ALLOWED_GB = 65
+FS_SIZE_METRIC = 'node_filesystem_size_bytes'
+FS_SIZE_METRIC_OLD = 'node_filesystem_size'
+AVAIL_SIZE_METRIC = 'node_filesystem_avail_bytes'
+AVAIL_SIZE_METRIC_OLD = 'node_filesystem_avail'
+
+
+class IcsSpaceAmplificationTest(IcsLongevetyTest):
+
+    def _get_used_capacity_gb(self, node) -> float:  # pylint: disable=too-many-locals
+        # node_filesystem_size_bytes{mountpoint="/var/lib/scylla", instance=~".*?10.0.79.46.*?"}-node_filesystem_avail_bytes{mountpoint="/var/lib/scylla", instance=~".*?10.0.79.46.*?"}
+        filesystem_capacity_query = f'{FS_SIZE_METRIC}{generate_capacity_query_postfix(node)}'
+        used_capacity_query = f'{filesystem_capacity_query}-{AVAIL_SIZE_METRIC}{generate_capacity_query_postfix(node)}'
+        self.log.debug(f"filesystem_capacity_query: {filesystem_capacity_query}")
+
+        fs_size_res = self.prometheus_db.query(query=filesystem_capacity_query, start=int(time.time()) - 5,
+                                               end=int(time.time()))
+        assert fs_size_res, "No results from Prometheus"
+        if not fs_size_res[0]:  # if no returned values - try the old metric names.
+            filesystem_capacity_query = f'{FS_SIZE_METRIC_OLD}{generate_capacity_query_postfix(node)}'
+            used_capacity_query = f'{filesystem_capacity_query}-{AVAIL_SIZE_METRIC_OLD}{generate_capacity_query_postfix(node)}'
+            self.log.debug(f"filesystem_capacity_query: {filesystem_capacity_query}")
+            fs_size_res = self.prometheus_db.query(query=filesystem_capacity_query, start=int(time.time()) - 5,
+                                                   end=int(time.time()))
+
+        assert fs_size_res[0], "Could not resolve capacity query result."
+        self.log.debug("used_capacity_query: {}".format(used_capacity_query))
+        used_cap_res = self.prometheus_db.query(query=used_capacity_query, start=int(time.time()) - 5,
+                                                end=int(time.time()))
+        assert used_cap_res, "No results from Prometheus"
+        used_size_mb = float(used_cap_res[0]["values"][0][1]) / float(MB_SIZE)
+        used_size_gb = float(used_size_mb / 1024)
+        self.log.debug(
+            "The used filesystem capacity on node {} is: {} MB/ {} GB".format(node.public_ip_address, used_size_mb,
+                                                                              used_size_gb))
+        return used_size_gb
+
+    def _get_filesystem_available_size_list(self, node, start_time):
+        """
+        :param node:
+        :return:
+        """
+        available_capacity_query = f'{AVAIL_SIZE_METRIC}{generate_capacity_query_postfix(node)}'
```

Comment:
Please don't use f-string if this work is intend to backported to 2019.1

Reply:
done